docs/EVE-IMAGE-SOURCES.md

```diff
@@ -0,0 +1,616 @@
+      },
+```
+
+The git commit for the above source is given in the label `"org.opencontainers.image.revision": "12a6fb8d29a6b47f998cd077dfb92213f0e6a55f"`, i.e.
+you can get the same commit via `git checkout 12a6fb8d29a6b47f998cd077dfb92213f0e6a55f`.
+
+The above _should_ be the same as the current git commit.
+
+Further, you can validate that the current contents of `pkg/pillar` give the correct output, by running:
+
+```sh
+$ linuxkit pkg show-tag pkg/pillar
+lfedge/eve-pillar:17837a9fcd05c765e9a1f6707b2e48f0f1dd215b
+```
+
+The above `17837a9fcd05c765e9a1f6707b2e48f0f1dd215b` is the same git tree hash used in the container image tag,
+`lfedge/eve-pillar:17837a9fcd05c765e9a1f6707b2e48f0f1dd215b-amd64`.
+
+## Container sources to Content
+
+With the specific source directory and commit in hand for each container image, we turn to the various sources that are used.
+
+There are several types of content that are loaded used to create the container image.
+
+### Local content
+
+Local content includes anything in the same directory, which therefore is used in the
+[docker build context](https://docs.docker.com/engine/reference/commandline/build/) when creating the container image.
+
+Everything used in the source directory is committed to git. This includes sources, the `Dockerfile` itself, and the `build.yaml`
+file used to control the image build via `linuxkit pkg build`.
+
+Further, `linuxkit pkg build` is used, it validates that the git repository is "clean", i.e. that there are no uncommitted changes
+or files not saved in git. If there are, it treats the repository as "dirty", and appends both `dirty` and a hash of all of the file
+contents in the directory to the tag. Thus `lfedge/eve-pillar:17837a9fcd05c765e9a1f6707b2e48f0f1dd215b-amd64` means that it checked,
+and there were no uncommitted changes or files in the source directory. If there were, it would have indicated on the tag as
+`lfedge/eve-pillar:17837a9fcd05c765e9a1f6707b2e48f0f1dd215b-dirty-<hash_of_file_contents>-amd64`.
+
+### Go modules
+
+Most of the compiled content used in EVE is written in [go](http://golang.org). For all go package dependencies, all sources
+use the [go module system](https://go.dev/blog/using-go-modules).
+
+The list of each module used is available in `go.mod`, including the [semantic version (semver)](https://semver.org) if the module
+uses semver, a hash otherwise. For example, [pkg/pillar/go.mod](../pkg/pillar/go.mod) contains partially:
+
+```go
+require (
+	cloud.google.com/go/storage v1.21.0 // indirect
+	github.com/Focinfi/go-dns-resolver v1.0.0
+	github.com/anatol/smart.go v0.0.0-20220615232124-371056cd18c3
+	github.com/bicomsystems/go-libzfs v0.4.0
+```
+
+Some of the above include only semver, while one includes the precise git commit and date or commit.
+
+Further, go modules preserve the actual hashes in [go.sum](../pkg/pillar/go.sum), including those for which semver was used.
```

Comment:
It might be useful to add something about how to take a line in go.sum and find the actual commit of the source code used.

Reply:
OK, added some more. Have a look.

Reply:
Looks good to me. I don't know if this is sufficient to create a complete list of URLs to all source. @ruslan-zededa are you looking at that? We can chat offline.